Feature/tictactoegame

NewIncome opened the pull request:
Tic Tac Toe game
We did:
- Created modules and factories for the players, gameBoard, gameFlow, render to the DOM
- Build functions to allow players to add marks
- Allow 2 players to put in their names
- Include buttons to start and restart the game
- Display a message to congratulate the winning player

bertil291utn committed:
Ganeboard and player object
- Create first two objects in order to achieve tic tac to game

NewIncome committed:
Add GameFlow module and test code.

bertil291utn committed:
New functionalities to Gameflow module
- Fix AddPlayer function
- Add makeMove function

NewIncome committed:
Add the gameOutcome function to know the result of the game.

bertil291utn committed:
Add winner function
- Test with draw and winner cases

bertil291utn committed:
Add logic function to initialize to play

NewIncome committed:
Continue testing with the game logic
Add past logic file for more tests

NewIncome committed:
Add a preliminary HTML and CSS board.
Add a Plan B test for the logic.

bertil291utn committed:
Update gameboard
- Add signs to check font size
- Hover to check if it's a block
- Add a handwrite font

bertil291utn committed:
Get id value from gameborard DOM
- Add 'define-custom-id' attribute as id parameter for DOM gameboard

NewIncome committed:
Add DOM module for DOM manipulation.

bertil291utn committed:
Gameboard blocks
- Display gameboard values in DOM
- Disable when a gameboard block has already a value

NewIncome committed:
Implement the gameflow to the DOM and add the winner display.

NewIncome committed:
Lock board cells when game is over
Center score text.

bertil291utn committed:
Split adEventlistener callback
- Add new function called GameFlowDOM to split in another function
eventlistener callback

NewIncome committed:
Fix eslinter errors
Add form element and styles for player names

bertil291utn committed:
Start game button change background color

bertil291utn committed:
Get flowgame to DOM
- Get names from inpute element

bertil291utn committed:
Fix linter errors, javascript file

bertil291utn committed:
Add new font
- Update to new chalk font

bertil291utn committed:
Add newGame and exit buttons after ends the game

bertil291utn committed:
Add logic to enter and newgame buttons
- Restart gameboard
- Clear inputs

bertil291utn committed:
Stepcounter variable
- Reset stepcounter variable to one exit game function

bertil291utn committed:
Change delete players
- Replace undefined set player variables to delete property
- Add min height to show the status message

bertil291utn committed:
Create LICENSE

Create an MIT license file

bertil291utn committed:
Merge branch 'feature/tictactoegame' of https://github.com/bertil291utn/tic-tac-toe-js into feature/tictactoegame

bertil291utn committed:
Update README file

bertil291utn committed:
Update license directory file
 README file

bertil291utn committed:
Update linters.yml file with new csss-tre validator

shloch reviewed (changes requested):
## Hello @NewIncome , @bertil291utn 

Great job....
Works perfectly when I play normally

![tic](https://user-images.githubusercontent.com/40712459/87359625-177fb880-c560-11ea-8aa9-745106f81efa.gif)
### [changes required]

- if I play in not-normal fashion, by click an already clicked square, I get something unusual, please fix that
![tic2](https://user-images.githubusercontent.com/40712459/87359692-3aaa6800-c560-11ea-9232-147f02bcc144.gif)

-  separate LOGIC from DOM ( see [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) }
You can split them into 2 js files and include then inside index.html
![code](https://user-images.githubusercontent.com/40712459/87359780-704f5100-c560-11ea-8c07-84d3a3c920a4.gif)

### Happy coding

NewIncome committed:
Fix error when clicking on an already clicked square
Separate logic from DOM manipulation, splited it into 2 files
Fix linter errors

Forison reviewed (approved):
## Status:: Approved✔️
# 🙋‍♂️ Perez & Bertil
### Good job on addressing the issues raised in the previous review🤗 .
### Go ahead and merge the branch and proceed to the next task🤗 .
#### Happy Coding👨‍💻
Addo Forison
